src/Infrastructure/Repository/TaskYamlRepository.php

```diff
@@ -455,4 +538,20 @@ protected function buildTaskSet(array $taskData, array $taskListData, array $tas
 
         return $task;
     }
+
+    /**
+     * @param string $taskId
+     *
+     * @throws \SprykerSdk\Sdk\Core\Application\Exception\TaskMissingException
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\TaskInterface
+     */
+    protected function getExistingTask(string $taskId): TaskInterface
+    {
+        if ($this->existingTasks[$taskId] instanceof TaskSetInterface) {
+            throw new TaskMissingException('Task set can\'t have another task set inside.');
```

Comment:
Would be fine to add taskId in exception message. I'm not sure that this check should be there. This value comes from the constructor and maybe put this check in constructor?

Reply:
*